uobtheatre/productions/models.py

```diff
@@ -232,6 +232,31 @@ def duration(self) -> Optional[datetime.timedelta]:
             return None
         return min(performance.duration() for performance in performances)
 
+    def sales_breakdown(self):
+        """Generates a breakdown of the sales of this production"""
+        from uobtheatre.bookings.models import Booking
+
+        bookings = (
+            Booking.objects.filter(performance__production=self)
+            .prefetch_related("payments")
+            .all()
+        )
+        total_payments = (
+            bookings.aggregate(Sum("payments__value"))["payments__value__sum"] or 0
+        )
+        total_misc_costs = sum(
+            [
+                booking.misc_costs_value() if len(booking.payments.all()) else 0
+                for booking in bookings
+            ]
+        )
+
+        return {
+            "payments_value": total_payments,
+            "misc_costs_value": total_misc_costs,
+            "society_income": total_payments - total_misc_costs,
+        }
```

Comment:
Lets make these separate properties on the model. If needs be we can make the bookings with payments a cached property to avoid computing that multiple times. Similarly with other properties